### PR TITLE
Added support for more decision conditions in trees and ONNX conversion

### PR DESCRIPTION
**Pull request by grafail**
Resolve issue mentioned in #226

### CODE REVIEW

**Comment by ksaur** [comment]
Thank you so much for this contribution!  Can you please add a test case that shows this? Maybe similar to the one shown in the issue?  Thanks!

**Comment by grafail** [comment]
I verified that the example in #226 still triggers the issue (by using BRANCH_LT) and added a test for it.

File: tests/test_onnxml_decision_tree_converter.py [hunk 1]
```diff
@@ -252,6 +255,24 @@ def test_random_forest_multi_random(self):
...
+        n_total = 100
+        np.random.seed(0)
+        X = np.random.rand(n_total, n_features)
+        X = np.array(X, dtype=np.float32)
+        y = np.random.randint(2, size=n_total)
+
+        # Create XGBoost Model
+        model = xgb.XGBClassifier()
+        model.fit(X, y)
+        self._test_classifier(X, model, conversion_function=convert_xgboost)
```
Comment (interesaaat) [inline tests/test_onnxml_decision_tree_converter.py]: Is this actually required? I think that if we call `convert` directly, internally it should pick `convert_sklearn` or `convert_xgboost` automatically based on the model type.

Comment (grafail) [reply tests/test_onnxml_decision_tree_converter.py]: I think it is required, as the issue can only be recreated when converting via onnx-ml. But indeed it seems `convert_model` from `onnxmltools` can be used to auto select the right converter.

**Review by interesaaat — APPROVED** [review]

**Comment by interesaaat** [comment]
Thanks for the contribution!

**Pull request merged**